src/tracer.js

```diff
@@ -133,6 +133,11 @@ export const trace = ({
     eventFilter,
   });
 
+  if (context.wrapped_by_lumigo) {
```

Comment:
I think you will have a bug here, try to test this on a non-async function
and don't call the callback immediately. 

Just something to let you think of, the user create a non-async func and you return him an async func

Reply:
I'm not sure what you meant,
I added a test and it worked.

Can we talk about it? :)